twilight-model/src/application/interaction/application_command/option.rs

```diff
@@ -83,16 +83,18 @@ impl<'de> Deserialize<'de> for CommandDataOption {
             Focused,
         }
 
-        // Id before string such that IDs will always be interpreted
-        // as such, this does mean that string inputs that looks like
-        // IDs will have to be caught if it is a string.
+        // An `Id` variant is purposely not present here.
+        // We first deserialize into a `String` and then
+        // later parse it into an `Id` variant. This is done
+        // to prevent situations where an option's string value
+        // is parsed as an `Id` variant, which omits features leading zeros.
+        // The `Id` variant is only used when the type is an `Id`.
```

Comment:
```suggestion
        // An `Id` variant is purposely not present here to prevent wrongly
        // parsing string options as numbers, trimming leading zeroes.
```